neptune/observation.go

```diff
@@ -2,13 +2,59 @@ package neptune
 
 import (
 	"context"
+	"errors"
+	"fmt"
+	"strings"
 
+	"github.com/ONSdigital/dp-graph/neptune/query"
 	"github.com/ONSdigital/dp-graph/observation"
 	"github.com/ONSdigital/dp-observation-importer/models"
 )
 
+// ErrEmptyFilter is returned if the provided filter is empty.
+var ErrEmptyFilter = errors.New("filter is empty")
+
 func (n *NeptuneDB) StreamCSVRows(ctx context.Context, filter *observation.Filter, limit *int) (observation.StreamRowReader, error) {
-	return nil, nil
+	if filter == nil {
+		return nil, ErrEmptyFilter
+	}
+
+	q := fmt.Sprintf(query.GetInstanceHeader, filter.InstanceID)
+
+	q += buildObservationsQuery(filter)
+	q += query.GetObservationSelectRowPart
+
+	if limit != nil {
+		q += fmt.Sprintf(query.LimitPart, *limit)
```

Comment:
I cannot reconcile the function signature taking a pointer (i.e. read/write) *limit* parameter, when the function provides only a thing from which data can subsequently be read. Could it be that we are obliged to
store the limit pointer in the cursor and write back to it as the subsequent reading operations take place?

Reply:
If we can bulid the query with limitation up front it will prevent unneccesary load on the database. If we don't limit this query the database could be attempting to load thousands of results, when we only care to read the first 10 and then close the query anyway.